### PR TITLE
Detect Docker Desktop (Docker for Mac) and warn about architecture emulation

### PR DESCRIPTION
**Pull request by pdcastro**
Connects-to: #1415
Change-type: patch
Signed-off-by: Paulo Castro <paulo@balena.io>

### CODE REVIEW

File: lib/utils/qemu-ts.ts [hunk 1]
```diff
@@ -42,20 +46,39 @@ export async function installQemuIfNeeded(
...
+	const isDockerDesktop = /(?:Docker Desktop)|(?:Docker for Mac)/i.test(
+		dockerInfo.OperatingSystem,
+	);
+	if (isDockerDesktop) {
+		logger.logInfo(stripIndent`
+			Docker Desktop detected (daemon architecture: "${dockerInfo.Architecture}")
+			Docker itself will determine and enable architecture emulation if required,
+			without balena-cli intervention and regardless of the --emulated option.`);
+	}
+	return !isDockerDesktop;
```
Comment (pdcastro) [inline lib/utils/qemu-ts.ts]: Sample build output with these changes:
```
$ ./bin/balena build -a test-rpi simple_container --logs --nocache --emulated
[Info]    Creating default composition with source: simple_container
[Info]    Building for armv7hf/raspberrypi3
[Build]   Building services...
[Build]   main Preparing...
[Info]    Docker Desktop detected (daemon architecture: "x86_64")
[Info]    Docker itself will determine and enable architecture emulation if required,
[Info]    without balena-cli intervention and regardless of the --emulated option.
[Build]   main Step 1/3 : FROM arm32v7/busybox
[Build]   main  ---> 723ce4685637
[Build]   main Step 2/3 : RUN uname -a
[Build]   main  ---> Running in f54620a94693
[Build]   main Linux f54620a94693 4.9.184-linuxkit #1 SMP Tue Jul 2 22:58:16 UTC 2019 armv7l GNU/Linux
```

Comment (CameronDiver) [reply lib/utils/qemu-ts.ts]: Generally what I tend to with these multiline message is to add a tab or a couple of spaces for every line that isn't the first, this seems to make things more clear that it's part of the same message.

**Comment by pdcastro** [comment]
force push = rebase on master

File: lib/utils/qemu-ts.ts (hunk 1 above)
Comment (pdcastro) [reply lib/utils/qemu-ts.ts]: Good idea. I've added a couple of spaces:
```
[Info]    Docker Desktop detected (daemon architecture: "x86_64")
[Info]      Docker itself will determine and enable architecture emulation if required,
[Info]      without balena-cli intervention and regardless of the --emulated option.
```

Comment (CameronDiver) [reply lib/utils/qemu-ts.ts]: Looks good, great work.

**Pull request merged**